### PR TITLE
Quad-SDK Soft Release

### PR DESCRIPTION
**Pull request by jcnorby**
Implements the following changes:

- Refactors of most of the major packages
- Updated documentation (especially readmes)
- Tuned global planner parameters
- Deprecation of old and unsupported packages

### CODE REVIEW

**Review by jrenaf — APPROVED** [review]

File: global_body_planner/include/global_body_planner/fast_global_motion_planner.h [hunk 1]
```diff
@@ -97,10 +92,9 @@ class RRTConnectClass : public RRTClass {
    * @param[out] action_sequence The sequence of actions in the final path
    * @return Boolean for success of the planner
    */
-  int runRRTConnect(const PlannerConfig &planner_config, State s_start,
-                    State s_goal, std::vector<State> &state_sequence,
-                    std::vector<Action> &action_sequence,
-                    ros::Publisher &tree_pub);
+  int findPlan(const PlannerConfig &planner_config, State s_start, State s_goal,
```
Comment (ardalantj) [inline global_body_planner/include/global_body_planner/fast_global_motion_planner.h:95]: should start and goal states be passed by reference?

Comment (jcnorby) [reply global_body_planner/include/global_body_planner/fast_global_motion_planner.h:95]: Yes, although that will be a future PR into devel. This PR should not include any new changes to keep main and devel conflict-free.

File: quad_utils/CMakeLists.txt [hunk 2]
```diff
@@ -61,11 +61,11 @@ add_library(quad_utils
   src/rviz_interface.cpp
   src/fast_terrain_map.cpp
   src/terrain_map_publisher.cpp
+  src/trajectory_publisher.cpp
```
Comment (ardalantj) [inline quad_utils/CMakeLists.txt:64]: I thought this was removed?

File: quad_utils/include/quad_utils/math_utils.h [hunk 3]
```diff
@@ -76,9 +75,9 @@ inline std::vector<double> wrapToPi(std::vector<double> data) {
  * @param[in] input_val Query point
  * @return Vector of interpolated values
  */
-std::vector<double> interpMat(std::vector<double> input_vec,
-                              std::vector<std::vector<double>> output_mat,
-                              double query_point);
+std::vector<double> interpMat(const std::vector<double> input_vec,
```
Comment (ardalantj) [inline quad_utils/include/quad_utils/math_utils.h:78]: missing &

File: quad_utils/include/quad_utils/math_utils.h [hunk 4]
```diff
@@ -89,9 +88,9 @@ std::vector<double> interpMat(std::vector<double> input_vec,
  * @param[in] input_val Query point
  * @return Vector of interpolated values
  */
-Eigen::Vector3d interpVector3d(std::vector<double> input_vec,
-                               std::vector<Eigen::Vector3d> output_mat,
-                               double query_point);
+Eigen::Vector3d interpVector3d(const std::vector<double> input_vec,
```
Comment (ardalantj) [inline quad_utils/include/quad_utils/math_utils.h:91]: missing &

File: quad_utils/include/quad_utils/math_utils.h [hunk 5]
```diff
@@ -103,8 +102,9 @@ Eigen::Vector3d interpVector3d(std::vector<double> input_vec,
  * @return Vector of interpolated values
  */
 std::vector<Eigen::Vector3d> interpMatVector3d(
-    std::vector<double> input_vec,
-    std::vector<std::vector<Eigen::Vector3d>> output_mat, double query_point);
+    const std::vector<double> input_vec,
```
Comment (ardalantj) [inline quad_utils/include/quad_utils/math_utils.h:105]: missing &

File: quad_utils/include/quad_utils/ros_utils.h [hunk 6]
```diff
@@ -196,15 +196,16 @@ void interpRobotPlan(quad_msgs::RobotPlan msg, double t,
...
-/**
- * @brief Interpolate data from a robot state trajectory message.
- * @param[in] msg robot state trajectory message
- * @param[in] t Time since beginning of trajectory (will return last state if
- * too large)
- * @return Robot state message
- */
-quad_msgs::RobotState interpRobotStateTraj(quad_msgs::RobotStateTrajectory msg,
-                                           double t);
+// /**
```
Comment (ardalantj) [inline quad_utils/include/quad_utils/ros_utils.h:199]: remove?

**Review by ardalantj — APPROVED** [review]

**Pull request merged**